### PR TITLE
ensure notebook-testing script fails when notebooks fail

### PR DESCRIPTION
**Pull request by jameslamb**
## Description

The notebook-testing CI job in this repo does not actually cause a loud CI failure if any errors are detected in notebooks. That's because of a `bash` mistake I made in #1407... that PR moved notebook-checking into a function, but didn't add a `set -E` to be sure errors from inside that function were appropriately trapped.

This PR fixes that:

* ensures that notebook failures actually cause CI failures
* fixes 2 typos in `nyc_taxi_years_correlation.ipynb` code
  - *(not caught in #1422 because of this CI script bug)*

Context: https://github.com/rapidsai/cuspatial/pull/1422#pullrequestreview-2212919011

### How I tested this

Originally did not skip any notebooks. Saw the failures in one of the notebooks cause an actual merge-blocking CI failure.

Build link: https://github.com/rapidsai/cuspatial/actions/runs/10199784404/job/28219162698?pr=1424

### Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: ci/test_notebooks.sh
Comment (jameslamb) [inline ci/test_notebooks.sh:32]: proposing skipping this until https://github.com/rapidsai/cuspatial/issues/1426 is resolved

**Comment by jameslamb** [comment]
This is ready for review. The failing unit test jobs are unrelated... they're the result of #1427, which will hopefully be fixed by @mroeschke 's PR: #1428

**Review by raydouglass — APPROVED** [review]

**Review by harrism — APPROVED** [review]

File: ci/test_notebooks.sh
Comment (jakirkham) [reply ci/test_notebooks.sh:32]: Now that the aforementioned issue has been resolved, can we please readd zipcodes?

```suggestion
SKIPNBS="binary_predicates.ipynb cuproj_benchmark.ipynb"
```

**Review by jakirkham — COMMENTED** [review]
Thanks James! 🙏

Looks good. Had one minor suggestion

File: ci/test_notebooks.sh
Comment (jameslamb) [reply ci/test_notebooks.sh:32]: ah yes! good catch, totally agree.

**Comment by jakirkham** [comment]
There were a couple CI jobs that appear to have had issues starting up (this is well before anything cuSpatial related AFAICT)

Have gone ahead and restarted those failing jobs. Hopefully that clears out the issue

**Review by jakirkham — APPROVED** [review]

**Comment by jameslamb** [comment]
Thanks @jakirkham !

I think we'll need #1434 to be finished to unblock CI in this repo.

**Comment by jameslamb** [comment]
One the notebooks is failing with what looks like a new `cudf`-compatibility issue. Wrote that up here: https://github.com/rapidsai/cuspatial/issues/1437

**Comment by jakirkham** [comment]
Going to covert to draft for clarity while those other blockers get cleared out

**Comment by harrism** [comment]
I think draft should be reserved for PRs unready for review. This PR is already reviewed, but can't be merged. I suggest using the DO NOT MERGE label, which does what it says on the tin.

**Comment by jameslamb** [comment]
> I think draft should be reserved for PRs unready for review.

No problem, I've moved this back to `Ready for Review` and added the `DO NOT MERGE` label.

**Comment by jameslamb** [comment]
Hopefully CI here will pass once #1438 is merged.

**Comment by jameslamb** [comment]
Thanks so much for all the help, everyone! Sorry this took a couple iterations, but I think this testing is in a better state now.

**Pull request merged**
